### ISSUE TITLE
Error: Length of 'hidden_states' different everytime

### ISSUE THREAD

**Issue by tarun360** [s3prl/s3prl#289]
Suppose I run the following code multiple times:
```
upstream = getattr(hub, 'wav2vec2')().to(device)
wavs = [torch.zeros(32000, dtype=torch.float).to(device) for _ in range(1)] # list of unpadded wavs `[wav1, wav2, ...]`, each wav is in `torch.FloatTensor`
with torch.no_grad():
    reps = upstream(wavs) # list of unpadded representations `[rep1, rep2, ...]`, each rep is of the shape `(extracted_seqlen, feature_dim)`
print(len(reps['hidden_states']))
```
Then the output is different every time. It ranges from 10-16. However, since wav2vec2 has 12 encoder layers, shouldn't it print 12 every time? In any case, I think the length of hidden_states shouldn't vary, it should be fixed. Do let me know if I am misunderstanding something.

**Comment by tarun360** [comment]
For example in of the runs when I print the dictionary, it prints as below:
```
 '_hidden_states_info': ('self.model.encoder.layers[0]',
  'self.model.encoder.layers[1]',
  'self.model.encoder.layers[3]',
  'self.model.encoder.layers[4]',
  'self.model.encoder.layers[6]',
  'self.model.encoder.layers[7]',
  'self.model.encoder.layers[8]',
  'self.model.encoder.layers[9]',
  'self.model.encoder.layers[11]',
  'self.model.encoder')
```
It's missing `self.model.encoder.layers[2]` and `self.model.encoder.layers[5]`

**Comment by leo19941227** [comment]
Hi,

This is a known situation.
You can refer to #192
To avoid this, simply turn the model into eval mode
upstream.eval()
